text/0000-cargo-target-directories.md

```diff
@@ -0,0 +1,283 @@
+-->
+
+## Do nothing
+
+It is already possible today to use `CARGO_TARGET_DIR` to remap workspaces and projects but this has two problems:
+
+1) If done globally, the `CARGO_TARGET_DIR` becomes a hodge-podge of every project, which is not often the goal.
+2) If done per-project, it is very cumbersome to maintain.
+
+For those reason, this option has not been retained.
+
+## Using `XDG_CACHE_HOME` instead of a cargo-specific env-var
+
+Not all OSes use the XDG convention, notably Windows and macOS (though the latter can be somewhat made to) and it is
+very easy to define `CARGO_TARGET_DIRECTORIES=${XDG_CACHE_HOME:-~/.cache}/cargo_target_directories` if wanted by users.
+
+## Just put the directories inside of `.cargo/cache/...`
+
+There are already lots of discussion about `.cargo` and `.rustup` being home to both cache and config files and why this
+is annoying for lots of users. What's more, it would not be as helpful to external build tools, they don't care about
+bringing the registry cache in their build directory for example.
+
+## Stabilize the naming scheme
+
+I feel this require an hard-to-break naming scheme, which I don't have the skills nor motivation to design. Instead, I prefer explicitely telling the naming scheme is not to
+be considered stable and allow more invested people to experiment with the feature and find something solid.
+
+# Prior art
+[prior-art]: #prior-art
+
+<!--
+Discuss prior art, both the good and the bad, in relation to this proposal.
+A few examples of what this can include are:
+
+- For language, library, cargo, tools, and compiler proposals: Does this feature exist in other programming languages and what experience have their community had?
+- For community proposals: Is this done by some other community and what were their experiences with it?
+- For other teams: What lessons can we learn from what other communities have done here?
+- Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+
+This section is intended to encourage you as an author to think about the lessons from other languages, provide readers of your RFC with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other languages.
+
+Note that while precedent set by other languages is some motivation, it does not on its own motivate an RFC.
+Please also take into consideration that rust sometimes intentionally diverges from common language features.
+-->
+
+- [`targo`](https://github.com/sunshowers/targo) by @sunshowers
+- [rust-lang/cargo#11156](https://github.com/rust-lang/cargo/issues/11156)
+
+Both express the needs for either remapping or a global target directory that is not shared between different Cargo workspaces.
```

Comment:
I feel like these more appropriately fall under "alternatives" and we should explain why the RFC is preferring this solution over the solutions in those especially since those solutions do not have collisions issues.

Prior art would be more about finding how other build tools handle this problem and comparing the RFC to them.

Reply:
Sorry for the delay, done !